File: psychology-education-backend/src/main/java/com/psychology/service/AdminService.java
package com.psychology.service;

import java.io.IOException;

import org.springframework.web.multipart.MultipartFile;

import com.psychology.dto.CourseDTO;
import com.psychology.dto.PsychologistDTO;
import com.psychology.dto.TopicDTO;
import com.psychology.entity.Course;
import com.psychology.entity.Psychologist;
import com.psychology.entity.Record;
import com.psychology.entity.Topic;
import com.psychology.exception.CourseAlreadyExistsException;
import com.psychology.exception.NotFoundException;
import com.psychology.exception.UserAlreadyExistsException;

public interface AdminService {
	Iterable<Psychologist> getPsychologists();

	Iterable<Course> getCourses();
	
	Iterable<Record> getRecords();
	
	Record activateRecord(Long id) throws NotFoundException;
	
	Record blockRecord(Long id) throws NotFoundException;
	
	Course getCourseById(Long id) throws NotFoundException;
	
	Topic getTopicById(Long id) throws NotFoundException;

	Long deletePsychologist(Long id) throws NotFoundException;

	Long deleteCourse(Long id) throws NotFoundException;

	Long deleteTopic(Long id) throws NotFoundException;

	Psychologist addPsychologist(PsychologistDTO psychologistDTO, MultipartFile file)
			throws UserAlreadyExistsException, IOException;

	Course addCourse(CourseDTO courseDTO, MultipartFile file) throws CourseAlreadyExistsException, IOException;
	
	Course addTopic(Long courseId, TopicDTO topicDTO) throws NotFoundException;
	
	Course updateCourse(CourseDTO courseDTO, MultipartFile file) throws CourseAlreadyExistsException, IOException;

	Topic updateTopic(TopicDTO topicDTO, Long id) throws NotFoundException;
}

File: psychology-education-frontend/src/API/AdminService.js
import axios from "axios";
import getAuthHeader from "Utils/getAuthHeader";

const API_URL = "http://localhost:8080/api/admin/";

export default class AdminService {
    static async getPsychologists() {
        const response = await axios.get(API_URL + "psychologists", {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getCourses() {
        const response = await axios.get(API_URL + "courses", {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getRecords() {
        const response = await axios.get(API_URL + "records", {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getCourseById(id) {
        const response = await axios.get(API_URL + `courses/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async activateRecords(id) {
        const response = await axios.get(API_URL + `records/activate/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async blockRecords(id) {
        const response = await axios.get(API_URL + `records/block/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getTopicById(id) {
        const response = await axios.get(API_URL + `topics/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getPsychologistAnalitic(id) {
        const response = await axios.get(API_URL + `analitics/psychologist/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getCoursesRating() {
        const response = await axios.get(API_URL + `analitics/courses`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async deletePsychologist(id) {
        const response = await axios.delete(API_URL + `psychologists/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async deleteCourse(id) {
        const response = await axios.delete(API_URL + `courses/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async deleteTopic(id) {
        const response = await axios.delete(API_URL + `topics/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addPsychologist(formData) {
        const response = await axios.post(API_URL + "psychologists", formData, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addCourse(formData) {
        const response = await axios.post(API_URL + "courses", formData, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addTopic({topic}, id) {
        const response = await axios.put(API_URL + `topics/${id}`, topic, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async updateCourse(formData) {
        const response = await axios.patch(API_URL + "courses", formData, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async updateTopic({topic}, id) {
        const response = await axios.patch(API_URL + `topics/${id}`, topic, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getProgressReport(id) {
        const response = await axios.get(API_URL + `report/${id}`, {
            headers: getAuthHeader(),
            responseType: 'arraybuffer'
        });
        return response;
    }
}
